src/__experimental__/components/checkable-input/hidden-checkable-input.spec.js

```diff
@@ -1,21 +1,119 @@
+      <InputContext.Provider value={ inputContextValue }>
+        <HiddenCheckableInput { ...props } />
+      </InputContext.Provider>
+    </InputGroupContext.Provider>
+  );
+};
 
 describe('HiddenCheckableInput', () => {
+  let propOnBlur;
+  let contextOnBlur;
+  let groupContextOnBlur;
+
+  let propOnFocus;
+  let contextOnFocus;
+  let groupContextOnFocus;
+
+  let contextOnMouseEnter;
+  let groupContextOnMouseEnter;
+
+  let contextOnMouseLeave;
+  let groupContextOnMouseLeave;
+
+  let wrapper;
+
+  beforeEach(() => {
+    propOnBlur = jest.fn();
+    contextOnBlur = jest.fn();
+    groupContextOnBlur = jest.fn();
+
+    propOnFocus = jest.fn();
+    contextOnFocus = jest.fn();
+    groupContextOnFocus = jest.fn();
+
+    contextOnMouseEnter = jest.fn();
+    groupContextOnMouseEnter = jest.fn();
+
+    contextOnMouseLeave = jest.fn();
+    groupContextOnMouseLeave = jest.fn();
+
+    wrapper = render(
+      {
+        onBlur: propOnBlur,
+        onFocus: propOnFocus
+      },
+      {
+        onBlur: groupContextOnBlur,
+        onFocus: groupContextOnFocus,
+        onMouseEnter: groupContextOnMouseEnter,
+        onMouseLeave: groupContextOnMouseLeave
+      },
+      {
+        onBlur: contextOnBlur,
+        onFocus: contextOnFocus,
+        onMouseEnter: contextOnMouseEnter,
+        onMouseLeave: contextOnMouseLeave
+      },
+    );
+  });
+
   it('renders as expected', () => {
-    expect(render({
+    const props = {
       checked: true,
       helpId: 'test-help',
       name: 'test-name',
       inputType: 'test-type',
       inputValue: 'test-value',
       tabindex: 0
-    })).toMatchSnapshot();
+    };
+
+    const shallowWrapper = shallow(<HiddenCheckableInput { ...props } />);
+    expect(shallowWrapper).toMatchSnapshot();
+  });
+
+  it('triggers onFocus callbacks passed from props and context', () => {
+    wrapper.find(HiddenCheckableInputStyle).props().onFocus();
+    expect(propOnFocus).toHaveBeenCalled();
+    expect(contextOnFocus).toHaveBeenCalled();
+    expect(groupContextOnFocus).toHaveBeenCalled();
+  });
+
+  it('triggers onBlur callbacks passed from props and context', () => {
+    wrapper.find(HiddenCheckableInputStyle).props().onBlur();
+    expect(propOnBlur).toHaveBeenCalled();
+    expect(contextOnBlur).toHaveBeenCalled();
+    expect(groupContextOnBlur).toHaveBeenCalled();
+  });
+
+  it('triggers onMouseEnter callbacks passed from context', () => {
+    wrapper.find(HiddenCheckableInputStyle).props().onMouseEnter();
+    expect(contextOnMouseEnter).toHaveBeenCalled();
+    expect(groupContextOnMouseEnter).toHaveBeenCalled();
+  });
+
+  it('triggers onMouseLeave callbacks passed from context', () => {
+    wrapper.find(HiddenCheckableInputStyle).props().onMouseLeave();
+    expect(contextOnMouseLeave).toHaveBeenCalled();
+    expect(groupContextOnMouseLeave).toHaveBeenCalled();
+  });
+
+  it('nothing happens if onFocus, onBlur, onMouseEnter or onMouseLeave callbacks are not provided', () => {
```

Comment:
Not sure this makes sense to me. Something like `it('does nothing if onFocus, onBlur, onMouseEnter or onMouseLeave callbacks are not provided')` would make more sense.